docs/rfcs/blocks-in-widget-areas.md

```diff
@@ -0,0 +1,520 @@
+		",
+		"rendered": ...
+	}
+}
+```
+
+This triggers a migration. `'footer'` now points to a numeric ID instead of an array. Other widget areas are not affected.
+
+```php
+array(
+	'footer' => 123,
+	'sidebar' => array(
+		'recent-posts-2',
+		'recent-comments-2',
+	),
+	'array_version' => 4,
+)
+```
+
+Here, `123` is the ID of a `wp_area` post. The `post_content` field of `get_post( 123 )` contains the `'footer'` widget area's block markup.
+
+```html
+<!-- wp:paragraph --><p>Hello there!</p><!-- /wp:paragraph -->
+<!-- wp:legacy-widget {"identifier":"search-2","instance":{"title":""}} /-->
+<!-- wp:legacy-widget {"identifier":"recent-posts-2","instance":{"title":"","number":5}} /-->
+<!-- wp:legacy-widget {"identifier":"recent-comments-2","instance":{"title":"","number":5}} /-->
+```
+
+## Backwards Compatibility
+
+### `wp_get_sidebars_widgets()`
+
+[`wp_get_sidebars_widgets()`][wp_get_sidebars_widgets] can now potentially return an associative array that maps strings to integers. This differs from the current behaviour which is to always return an associative array that maps strings to arrays.
+
+Since `wp_get_sidebars_widgets()` is marked `@private`, no effort is made to ensure backwards compatibility beyond incrementing the `'array_version'` field from `3` to `4`.
+
+The same goes for [`wp_set_sidebars_widgets()`][wp_set_sidebars_widgets] and `get_option( 'sidebars_widgets' )`.
+
+[wp_get_sidebars_widgets]: https://developer.wordpress.org/reference/functions/wp_get_sidebars_widgets/
+[wp_set_sidebars_widgets]: https://developer.wordpress.org/reference/functions/wp_set_sidebars_widgets/
+
+### `'sidebars_widgets'` filter
+
+The [`'sidebars_widgets'` filter][sidebars_widgets] remains as-is and is passed an associative array that maps widget area identifiers to an an array of widget identifiers.
+
+To do this, sequences of blocks in a widget area will be converted into dynamically registered _blocks widgets_ which outputs the blocks in that sequence.
+
+To illustrate, let's say that the `'footer'` widget area has been migrated.
+
+```php
+array(
+	'footer' => 123,
+	'array_version' => 4,
+)
+```
+
+The referenced `wp_area` contains some blocks and a legacy widget in its `post_content`.
+
+```html
+<!-- wp:paragraph --><p>Hello there!</p><!-- /wp:paragraph -->
+<!-- wp:legacy-widget {"identifier":"search-2","instance":{"title":""}} /-->
+<!-- wp:separator --><hr class="wp-block-separator"/><!-- /wp:separator -->
+<!-- wp:paragraph --><p>Welcome to my cool website!</p><!-- /wp:paragraph -->
+```
+
+Now, let's take a look at what the `'sidebars_widgets'` filter receives.
+
+```php
+function my_sidebars_widgets_filter( $sidebars_widgets ) {
+	var_dump( $sidebars_widgets );
+	return $sidebars_widgets;
+}
+add_filter( 'sidebars_widgets', 'my_sidebars_widgets_filter' );
+```
+
+We receive an array that contains three widgets.
+
+```php
+array(
+	'footer' => array(
+		'wp-blocks-cb8d16f47bf5cfa6ef313e813d6b80d6',
```

Comment:
What if for back-compatibility we use a single widget that contains all the blocks or legacy widget blocks? e.g:
```
'footer' => array(
		'wp-blocks-footer',
)
```
It seems a simpler approach.

Reply:
We want to keep the e.g. `search-2` widget for backwards compatibility, though.